### ISSUE TITLE
dnf module install python3 fails

### ISSUE THREAD

**Issue by alexxa** [modularity-modules/python3#4]
Error: 
 Problem: conflicting requests
  - nothing provides python3-tkinter = 3.6.2-18.module_08d8dbe4 needed by python3-tools-3.6.2-18.module_08d8dbe4.x86_64

**Comment by alexxa** [comment]
@KarstenHopp

**Comment by torsava** [comment]
I'm looking at the issue, thanks for reporting!

**Comment by torsava** [comment]
@alexxa I rebuilt the module to fix this issue, could you please check the installability now?

And what approach do you recommend for testing the installability by myself? Thanks!

**Comment by alexxa** [comment]
@torsava copy from irc: 
python3 is installed, but not displayed by list --installed. mhatina said he would investigate

```
Boltron-27-765db245d6ef-0# dnf module list --installed                                                                                                    Loading repositories.
Last metadata expiration check: 0:29:30 ago on Wed 04 Oct 2017 12:41:55 PM UTC.
Fedora Modular 27 - x86_64
Name            Stream                   Version                Profiles                   Installed         Info                                         
python2         master (default)         20171003150125         default, full, ...         default           Python programming language, version 2       

Boltron-27-765db245d6ef-0# dnf module install python3
Loading repositories.
Last metadata expiration check: 0:29:49 ago on Wed 04 Oct 2017 12:41:55 PM UTC.
Package python3-3.6.2-19.module_34c36fd9.x86_64 is already installed, skipping.
Package python3-libs-3.6.2-19.module_34c36fd9.x86_64 is already installed, skipping.
Package python3-pip-9.0.1-12.module_34c36fd9.noarch is already installed, skipping.
Package python3-setuptools-36.2.0-7.module_34c36fd9.noarch is already installed, skipping.
Dependencies resolved.
Nothing to do.
Complete!
```

**Comment by torsava** [comment]
@alexxa Any info? I'm going to be unreachable for a week starting on Wednesday, so it would be best to solve this today or tomorrow.

**Comment by KarstenHopp** [comment]
If you have to do another rebuild, could you please request a 3.6 branch in modules/python3 dist-git and build it from there to get the correct stream name ?  Thanks !

**Comment by KarstenHopp** [comment]
like this:
fedrepo-req-branch -n modules --sl bug_fixes:2018-12-01 python3 3.6

**Comment by torsava** [comment]
> If you have to do another rebuild, could you please request a 3.6 branch in modules/python3 dist-git and build it from there to get the correct stream name ? Thanks !

Since we're not planning to support multiple Python 3 versions in F27, I don't think it would be a good idea to add the 3.6 branch. Especially since it would be just a copy of the master branch.

**Comment by ignatenkobrain** [comment]
@KarstenHopp also when 3.7 will be released, @torsava will have to create new package python3.6 because module streams are not parallel-installable.

**Comment by whitel** [comment]
How do you you get python3.6 on f*28*? Wouldn't i want to be able to bring my app that depends on 3.6 forward to f28 with confidence that it is the same stream?

**Comment by ignatenkobrain** [comment]
@whitel well, you don't. because our packaging supports only one version of python3 subpackages and it is what python3 package provides. Since we want py3.6, py3.7, ... to be parallel-installable, they have to be different package names and different modules

**Comment by torsava** [comment]
@ignatenkobrain That was the original plan, but since modularity people informed us that we can't do that, because other modules need to be able to be built against all the available Python 3 versions, and therefore the different Python 3 versions have to be streams of one module.

**Comment by ignatenkobrain** [comment]
@torsava so you give up on installing pythons in parallel?

**Comment by torsava** [comment]
@whitel @KarstenHopp I'm not opposed to having the main stream of the Python 3 module named 3.6, and it would certainly make sense. However, I am opposed to having two streams that we have to maintain with the same identical content. And the other modules I've seen are all using the master branch and not a versioned branch.

If it's not a problem to obsolete the `master` branch (can we even do it??) I suppose we could switch to a branch named `3.6` for the `python3` module. Though I would have to find someone to delegate it to since I'm gonna be off for a week starting on Wednesday and there are 6 Python modules that almost always fail to build due to something being changed since last time. :)

**Comment by torsava** [comment]
> @torsava so you give up on installing pythons in parallel?

We decided that we just cannot do it cleanly within the Modularity framework, and therefore if you want to use Modular Fedora you have to accept its limitations. Mainline Fedora, as long as it's supported, will have parallel installability of Pythons.

**Comment by ncoghlan** [comment]
Switching to a 3.6 branch makes sense to me, since that branch would then cover F28 in addition to F27.

A future 3.7 branch would then cover F29, F30, F31, and maybe F32.

**Comment by ncoghlan** [comment]
I filed a new issue for the renaming request though, since that doesn't seem to be related to the original problem reported here: https://github.com/modularity-modules/python3/issues/5

**Comment by ncoghlan** [comment]
Note: this issue should probably be closed, since installation currently works fine.

**Comment by torsava** [comment]
Closing, thanks @ncoghlan!
